src/state/slices/common-selectors.ts

```diff
@@ -85,3 +89,57 @@ export const selectPortfolioFiatBalances = createDeepEqualOutputSelector(
       return acc
     }, {}),
 )
+
+export const selectPortfolioFiatBalancesByAccount = createDeepEqualOutputSelector(
+  selectAssets,
+  selectPortfolioAccountBalancesBaseUnit,
+  selectMarketDataSortedByMarketCap,
+  (assetsById, accounts, marketData) => {
+    return Object.entries(accounts).reduce(
+      (acc, [accountId, balanceObj]) => {
+        acc[accountId] = Object.entries(balanceObj).reduce(
+          (acc, [assetId, cryptoBalance]) => {
+            const asset = assetsById[assetId]
+            if (!asset) return acc
+            const precision = asset.precision
+            const price = marketData[assetId]?.price ?? 0
+            const cryptoValue = fromBaseUnit(bnOrZero(cryptoBalance), precision)
+            const fiatBalance = bnOrZero(bn(cryptoValue).times(price)).toFixed(2)
+            acc[assetId] = fiatBalance
+
+            return acc
+          },
+          { ...balanceObj },
+        )
+
+        return acc
+      },
+      { ...accounts },
+    )
+  },
+)
+
+export const selectSortedAssets = createSelector(
+  selectAssets,
+  selectPortfolioFiatBalances,
+  selectPortfolioFiatBalancesByAccount,
+  selectMarketDataSortedByMarketCap,
+  (assets, portfolioFiatBalances, cryptoMarketData) => {
+    const selectAssetFiatBalance = (asset: Asset) =>
+      bnOrZero(portfolioFiatBalances[asset.assetId]).toNumber()
+    const selectAssetMarketCap = (asset: Asset) =>
+      bnOrZero(cryptoMarketData[asset.assetId]?.marketCap).toNumber()
+    const selectAssetName = (asset: Asset) => asset.name
+
+    return orderBy(
+      Object.values(assets).filter(isSome),
+      [selectAssetFiatBalance, selectAssetMarketCap, selectAssetName],
+      ['desc', 'desc', 'asc'],
+    )
+  },
+)
+
+export const selectSortedAssetsById = createSelector(
+  selectSortedAssets,
+  sortedAssets => new Map(sortedAssets.map(asset => [asset.assetId, asset])),
```

Comment:
this breaks the redux dev tools, and by extension react

everything returned from selectors needs to be serializable to JSON and we can't serialize maps, and view layers can't react correctly on a Map. can we just use `PartialRecord<AssetId, Asset>` here? we get the same `O(1)` lookups